src/style_manager/index.ts

```diff
@@ -597,27 +597,65 @@ export default class StyleManager extends ItemManagerModule<
       const cmp = target.toHTML ? target : target.getComponent();
       const optsSel = { array: true } as const;
       let cmpRules: CssRule[] = [];
+      let tagNameRules: CssRule[] = [];
+      let invisibleRules: CssRule[] = [];
       let otherRules: CssRule[] = [];
       let rules: CssRule[] = [];
 
       const rulesBySelectors = (values: string[]) => {
-        return !values.length
-          ? []
-          : cssC.getRules().filter(rule => {
-              const rSels = rule.getSelectors().map(s => s.getFullName());
-              return !!rSels.length && rSels.every(rSel => values.indexOf(rSel) >= 0);
+        return cssC.getRules().filter(rule => {
+          const rSels = rule.getSelectors().map(s => s.getFullName());
+
+          // rSels is equal to 0 when rule selectors contain tagName like : p {}, .logo path {}, ul li {}
+          if (rSels.length === 0) {
+            return false;
+          }
+
+          return rSels.every(rSel => values.indexOf(rSel) >= 0);
+        });
+      };
+
+      const rulesByTagName = (tagName: string) => {
+        return cssC.getRules().filter(rule => {
+          const ruleTags = rule
+            .selectorsToString()
+            .split(' ')
+            .filter(item => {
+              return item.startsWith('.') === false && item.startsWith('#') === false;
```

Comment:
why not to simply check `rule.selectorsToString() === tagName`?
The logic here will take wrong rules, eg. `.something DIV .something-else`

Reply:
I cannot just check `rule.selectorsToString() === tagName` because `tagName` only include the tag name like `div`, `section`, `li`... and not all the path like `rule.selectorsToString()` can have (`.something .something-else div`).

In fact, it's a problem for this case: `.something DIV .something-else`. So I add a check to be sure the tagName is the last element in the `rule.selectorsToString()`.

I know this function is not perfect, it still contains this caveat:

```html
<section class="cls1">
  <div>Text</div>
</section>
```
```css
.cls1 div {
  padding: 10px;
}
```

In the previous case,`.cls1 div` will be return as a rule for both div because I don't how to check a component is a child or not of a component having `.cls1`. Or if we can check that I think we will add a lot of loops...

I don't know what is the best things to do:

1 - Know the caveat and let it like this
2 - Try to fix it and add a lot of checks and loops

In every case, I fix the following case `.something DIV .something-else` and add a test on it :)

Reply:
I'd suggest to keep it simple for now, only selectors with one and single tagName are accepted, eg. `div` is ok, `.cls1 div` is not.  So let's just do `rule.selectorsToString() === tagName`.

You're not considering a lot of CSS selectors, like in your case `.cls2 div` will be taken as valid, also `.cls2 div div`... it's better to skip those rules instead of showing the wrong ones. 
We won't be able to be precise with the current approach anyway.

Reply:
The 0.17 version is supported.